pyproject.toml

```
@@ -5,7 +5,7 @@ build-backend = "setuptools.build_meta"
 [project]
 name = "chemprop"
 description = "Molecular Property Prediction with Message Passing Neural Networks"
-version = "1.5.1"
+version = "2.0.0a0"
```

Is this changing the pyproject.toml in the top level chemprop folder that still has v1? If so maybe we should wait to update this until we move v2 out of its folder and v1 into a folder.

---

It is indeed changing in the top level folder. I had to do it here since the pip install in the CI was expecting to use a file found in the root directory to do the install. I don't know off the top of my head if you are able to pip install a package if the pyrpoject.toml file is in a different directory.

---

I think this is fine since it's on the `v2/dev` branch. I don't think anyone will try to install v1 from this branch.

---

However, I think we should change the versioning to match the semantic versioning [specification](https://semver.org/) for pre-releases.

---

```suggestion
version = "2.0.0-alpha"
```

---

Based on my re-reading of the `semver` and `bump2version` docs, I think I agree with @davidegraff that we should keep this as `1.5.1` for now.

From the semver docs, I think `2.0.0-alpha` will take precedence over `1.5.1` since it doesn't compare pre-release versions unless major, minor, and patch versions are the same (does this mean someone installing `chemprop` with `pip` will get the alpha version instead of the stable version if they don't specify a version, assuming they have python 3.11?).

So should we only run `bump2version` and make a new PyPI release once `2.0.0` is stable?

---

pip will not install releases marked as "pre-releases" unless the user specifically allows it (with `--pre`), so I think in the case that only a pre-release would satisfy installation requirements (like here, if a user tried to install from 3.12) it would still not install (but I might be wrong). I'll add though that we haven't actually put anything on PyPI (and our automated build action doesn't run on this branch), so this isn't possible regardless.

2.0.0-alpha will be changed by python's `build` package to 2.0.0a0 (see [PEP440](https://peps.python.org/pep-0440/) for Python's supported version specifiers) which is why I had it that way.

I went ahead and changed it since I never use bumpversion personally, though I know it's popular. I think it's less work to just manually edit the single-sourced version in pyproject.toml rather than install a utility to do it.

---

IMO it's useful for when you have multiple sources of the version info (i.e., docs, package, etc.) that can't be "sourced" from the package info (at least I'm not familiar with a way to dynamically build my documentation such that it uses the package version as its source.) In any case, the `2.0.0a0` tag already exists (a minimally functional version back from Nov. '22), so we can call this working branch `a1`.
```suggestion
version = "2.0.0a1"
```